modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -152,27 +140,126 @@ def run
     if @zoomeye_token.blank?
       print_error("Unable to login api.zoomeye.org")
       return
+    else
+      print_status("Logged in to zoomeye")
     end
 
-    # create ZoomEye request parameters
-    dork = datastore['ZOOMEYE_DORK']
-    resource = datastore['RESOURCE']
-    page = 1
-    maxpage = datastore['MAXPAGE']
+    results = []
+    results[0] = dork_search(resource, dork, 1, facets)
 
+    if results[0]['total'].nil? || results[0]['total'] == 0
+      msg = "No results."
+      if results[0]['error'].to_s.length > 0
+        msg << " Error: #{results[0]['error']}"
+      end
+      print_error(msg)
+      return
+    end
+
+    # Determine page count based on total results
+    if results[0]['total'] % 20 == 0
+      tpages = results[0]['total'] / 20
+    else
+      tpages = results[0]['total'] / 20 + 1
+    end
+    maxpage = tpages if datastore['MAXPAGE'] > tpages
+
+    print_status("Total: #{results[0]['total']} on #{tpages} " +
+      "pages. Showing: #{maxpage} page(s)")
+
+    # If search results greater than 20, loop & get all results
+    if results[0]['total'] > 20
```

Comment:
Can you update this to instead of constantly referring to `results[0]`, call it something like `initial_page`, `first_page` etc. Then at the end, to process all of the results, you can prepend it to the results array. This would help make the code a bit more readable.